### ISSUE TITLE
Factus: Introduce HandlerParameterResolver and define a lifecycle for it

### ISSUE THREAD

**Issue by uweschaefer** [factcast/factcast#1077]
It would be nice to extend the number of possible parameter types to @Handler methods by registering an appropriate bean to spring.
If we also define a lifecycle for object that are created and destroyed around the event processing, we can combine these two features to support transactionality without any spooky threadlocal, spring-proxy or similar mechanism.

something like 
```
@Handler
void apply(RTransaction redissonTransaction, FooEvent e)
```
comes to mind.

**Comment by uweschaefer** [comment]
maybe this illustrates the idea a bit more:
```java
public interface ProjectorSupport {
    default void beforeCatchup(Projection p) {
    }

    default void beforeProjecting(Projection p) {
    }

    default <T> T resolveParameter(Class<T> parameterType, Projection p, Fact f) {
        return null;
    }

    default void afterProjecting(Projection p, Throwable e) {
    }

    default void afterCatchup(Projection p, Throwable e) {
    }
}

@RequiredArgsConstructor
class RedissonTransactionalProjectorSupport implements ProjectorSupport {

    final Redisson redisson;

    IdentityHashMap<Projection, RTransaction> activeTransactions = new IdentityHashMap<Projection, RTransaction>();

    @SuppressWarnings("unchecked")
    @Override
    public <T> T resolveParameter(Class<T> parameterType, Projection p, Fact f) {
        if (RTransaction.class.equals(parameterType)) {
            return (T) activeTransactions.computeIfAbsent(p, projection -> redisson.createTransaction());
        }
        return null;
    }

    @Override
    public void afterProjecting(Projection p, Throwable e) {
        RTransaction tx = activeTransactions.remove(p);
        if (tx != null) {
            if (e != null) {
                tx.commit();
            }
            else {
                tx.rollback();
            }
        }
    }
}
```

**Comment by uweschaefer** [comment]
closed by #1325
